File: hanoi.clj
(defn hanoi
  ([]
    (hanoi (read-string (read-line)) "a" "b" "c"))

  ([n x y z]
    (when (> n 1)
      (hanoi (- n 1) x z y))
    (println (str x "=>" y))
    (when (> n 1)
      (hanoi (- n 1) z y x))))

(hanoi)

File: hanoi.lua
function hanoi (n, x, y, z)
  if n > 1 then hanoi(n - 1, x, z, y) end
  print(x .. "=>" .. y)
  if n > 1 then hanoi(n - 1, z, y, x) end
end

hanoi(tonumber(io.read()), "a", "b", "c")
